packages/cache/src/models/Subscribable.ts

```diff
@@ -0,0 +1,37 @@
+import type { CacheKey, CacheOptions, Sync } from '../types'
+import { hashCacheKey } from '../utils'
+
+export class Subscribable {
+  protected syncsMap: Map<ReturnType<typeof hashCacheKey>, Array<Sync>>
+
+  constructor() {
+    this.syncsMap = new Map()
+    this.subscribe = this.subscribe.bind(this)
```

Comment:
```ts
    this.subscribe = this.subscribe.bind(this)
```

[NIT]
Is it necessary to bind `this`?

Reply:
For example, when passing a method as a parameter to another function, the reference to this can change, causing this to point to something other than what was originally intended. In such cases, explicitly binding this ensures that it refers to the expected object.

## When I didn't bind it
<img width="401" alt="스크린샷 2024-08-15 오후 10 16 11" src="https://github.com/user-attachments/assets/f4fa8f7c-1762-4c7b-afbb-b17cd4d80e6a">

## When you bind it
<img width="475" alt="스크린샷 2024-08-15 오후 10 16 19" src="https://github.com/user-attachments/assets/918d596a-0e95-4924-9c97-5b9356110007">

Reply:
To elaborate further, the subscribe method can be passed as a callback function or assigned to a variable, which may change the reference. For this reason, we use binding to explicitly specify what this refers to.

Reply:
I hope that answer was helpful :)

Reply:
Oh I got it. Thanks!